File: backend/app/schemas/analytics.py
from datetime import date, datetime

from app.schemas.common import ORMModel


class TrackSummaryRead(ORMModel):
    track_id: int
    track_name: str
    code: str
    region: str
    total_races: int
    completed_races: int
    total_entries: int
    unique_players: int
    latest_race_date: date | None
    race_status_counts: dict[str, int]
    grade_counts: dict[str, int]
    recent_30_races: list[dict[str, object]]


class TrackPlayerStatRead(ORMModel):
    player_id: int
    player_number: int
    name: str
    grade: str
    starts: int
    wins: int
    top2: int
    top3: int
    win_rate: float
    top2_rate: float
    top3_rate: float
    disqualified_count: int
    withdrawn_count: int


class RaceSummaryRead(ORMModel):
    total_races: int
    scheduled_races: int
    in_progress_races: int
    completed_races: int
    total_players: int
    total_results: int
    latest_race_date: date | None
    track_count: int

File: frontend/lib/models/analytics.dart
class AnalyticsDashboardSummary {
  const AnalyticsDashboardSummary({
    required this.totalRaces,
    required this.scheduledRaces,
    required this.inProgressRaces,
    required this.completedRaces,
    required this.totalPlayers,
    required this.totalResults,
    required this.latestRaceDate,
    required this.trackCount,
  });

  final int totalRaces;
  final int scheduledRaces;
  final int inProgressRaces;
  final int completedRaces;
  final int totalPlayers;
  final int totalResults;
  final String? latestRaceDate;
  final int trackCount;

  factory AnalyticsDashboardSummary.fromJson(Map<String, dynamic> json) {
    return AnalyticsDashboardSummary(
      totalRaces: json['total_races'] as int,
      scheduledRaces: json['scheduled_races'] as int,
      inProgressRaces: json['in_progress_races'] as int,
      completedRaces: json['completed_races'] as int,
      totalPlayers: json['total_players'] as int,
      totalResults: json['total_results'] as int,
      latestRaceDate: json['latest_race_date'] as String?,
      trackCount: json['track_count'] as int,
    );
  }
}
